java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java

```diff
@@ -209,30 +209,61 @@ private static void updateLabelFromSecret(SecretManagerServiceClient client, Sec
+    private static void destroyOldSecretVersions(SecretManagerServiceClient client, SecretName secretName, SecretVersion exceptVersion) {
+
+        // 1. destroy up to ONE page of ENABLED secret versions (max 20)
         try {
-            SecretManagerServiceClient.ListSecretVersionsPage enabledVersions = client.listSecretVersions(ListSecretVersionsRequest.newBuilder()
+            SecretManagerServiceClient.ListSecretVersionsPage enabledVersions =
+                client.listSecretVersions(ListSecretVersionsRequest.newBuilder()
                             .setFilter("state:ENABLED")
                             .setParent(secretName.toString())
                             // Reduce the page, as each version will be disabled one by one
                             .setPageSize(NUMBER_OF_VERSIONS_TO_RETRIEVE)
                             .build())
                     .getPage();
 
-            // Disable secret version, just the first page
             enabledVersions.getValues().forEach(i -> {
                 if (!i.getName().equals(exceptVersion.getName())) {
-                    client.disableSecretVersion(DisableSecretVersionRequest.newBuilder()
+                    client.destroySecretVersion(DestroySecretVersionRequest.newBuilder()
                             .setName(i.getName())
                             .build());
 
-                    log.info(String.format("Property: %s, disabled version %s", secretName, i.getName()));
+                    log.info(String.format("Property: %s, destroyed version %s", secretName, i.getName()));
                 }
             });
         } catch (Exception e) {
-            log.log(Level.SEVERE, String.format("Cannot disable old versions from secret %s", secretName.toString()), e);
+            log.log(Level.SEVERE, String.format("Failed to destroy old versions of secret %s", secretName), e);
             throw e;
         }
+
+        // 2. transitionally, destroy up to ONE page of old, DISABLED secret versions
+        // transitional - added in 0.4.49; previously, we just DISABLED secret versions, but GCP
+        // still bills for these - "0.06 per version per location per month for 'Active secret versions'"
+        // in Notes section of their pricing paging,
+        // "1. A secret version is 'active' when it's in any of these states: ENABLED, DISABLED"
+       // so we need to DESTROY any versions of these secrets that are merely DISABLED
+        try  {
+            Instant until = Instant.ofEpochSecond(exceptVersion.getCreateTime().getSeconds())
+                .minus(Duration.ofDays(7));
+
+            SecretManagerServiceClient.ListSecretVersionsPage disabledVersions =
+                client.listSecretVersions(ListSecretVersionsRequest.newBuilder()
+                    .setFilter("state:DISABLED AND create_time < " + until.getEpochSecond() + "s")
```

Comment:
first time I see epochSecond as filter, but OK. Guess @aperez-worklytics  can test this on his proxy, I don't have a GCP one either

Reply:
Added this commit https://github.com/Worklytics/psoxy/pull/657/commits/ce769c9a3db1aaacb6b4cbe70b072f3f6772e46c @eschultink after trying it which fixes the filtering. Working on my env after that.